apps/nextjs/next.config.ts

```diff
@@ -32,15 +32,31 @@ const nextConfig: NextConfig = {
   experimental: {
     optimizePackageImports: ["@mantine/core", "@mantine/hooks", "@tabler/icons-react"],
     turbopackFileSystemCacheForDev: true,
+    // Reduce memory usage by limiting concurrent requests
+    serverActions: {
+      bodySizeLimit: "1mb",
+    },
+    // Reduce memory usage by limiting worker threads
+    workerThreads: false,
+    // Optimize memory usage
+    memoryBasedWorkersCount: true,
   },
+  // Reduce memory usage in production
+  compress: true,
```

Comment:
Can be removed as it represents the default value, see https://nextjs.org/docs/app/api-reference/config/next-config-js/compress